src/app/use-cases/request-signature.ts

```diff
@@ -71,3 +81,54 @@ export const makeRequestSignature =
       ),
       TE.chainFirst(addSignatureRequest)
     );
+
+/*
+ * Update status of signature request only if:
+ * new status is READY
+ * and previous status is WAIT_FOR_ISSUER!
+ */
+export const updateStatusRequestSignature =
+  (upsertSignatureRequest: UpsertSignatureRequest) =>
+  (
+    payload: RequestSignatureStatusPayload
+  ): TE.TaskEither<Error, SignatureRequest> =>
+    pipe(
+      sequenceS(TE.ApplySeq)({
+        signatureRequestStatus: pipe(
+          TE.of(payload.signatureRequestStatus),
+          TE.filterOrElse(
+            (_) => _ === "READY",
+            constant(new InvalidEntityError("Only READY status is allowed!"))
```

Comment:
you should use `ValidationError` (from handler-kit) instead `InvalidEntityError`

try to move this check to `io-ts` :)

Reply:
Actually you used `ValidationError` in this PR, in fact it is built by the `validate` function of `handler-kit`

Reply:
After creating a `io-ts` decoder that checks the `status`, you can receive a `ValidationError` by passing it to the `validate` function.

Reply:
OK thank you. Now it is clearer!

Reply:
t.Errors is the left result when calling `decode()`